opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -497,6 +512,78 @@ namespace Opm
                 }
         }
 
+        /// Create sparsity pattern for block-Jacobi matrix based on partitioning of grid.
+        void blockJacobiAdjacency()
+        {
+            const auto& grid = simulator_.vanguard().grid();
+            std::vector<int> cell_part = simulator_.vanguard().cellPartition();
+
+            typedef typename Matrix::size_type size_type;
+            size_type numCells = grid.size( 0 );
+            blockJacobiForGPUILU0_.reset(new Matrix(numCells, numCells, Matrix::random));
+
+            std::vector<std::set<size_type>> pattern;
+            pattern.resize(numCells);
+
+            const auto& lid = grid.localIdSet();
+            const auto& gridView = grid.leafGridView();
+            auto elemIt = gridView.template begin<0>();
+            const auto& elemEndIt = gridView.template end<0>();
+
+            //Loop over cells
+            for (; elemIt != elemEndIt; ++elemIt)
+            {
+
+                const auto& elem = *elemIt;
+                size_type idx = lid.id(elem);
+                pattern[idx].insert(idx);
+
+                // Add well non-zero connections
+                for (auto wc = wellConnectionsGraph_[idx].begin(); wc!=wellConnectionsGraph_[idx].end(); ++wc)
+                    pattern[idx].insert(*wc);
```

Comment:
This is interesting. If there are wells the domains might still be connected. Was not clear to me.

@andrthu How are you doing the decomposition? Are the wells allowed to be cut?